### PR TITLE
Add RELEASE_PROCESS.md file

### PR DESCRIPTION
**Pull request by crhino**
Document the Terraform module release process.

### CODE REVIEW

**Review by chapmanc — APPROVED** [review]
LGTM

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+...
+```
+
+2. Run `make generate_templates` from root level of repository. This should
+   update all examples and templates with the newest version.
+
+3. Submit a PR and get it reviewed and merged.
+
+4. Tag and push the new version to Github.
```
Comment (jjtimmons) [inline RELEASE_PROCESS.md:23]: I know we haven't done it lately repo, but I kind of like Github releases a bit. They'll make release notes and stuff which are cool for multiple PRs in the latest set of changes

**Review by jjtimmons — APPROVED** [review]
Is dope

**Comment by jjtimmons** [comment]
@crhino I'm doing a few things in this repo ATM so going to merge this one as well

**Pull request merged**